### PR TITLE
Refactor waiter handling in channel

### PR DESCRIPTION
**Pull request by foolswood**
Rough pass at tidying up waiter handling.

### CODE REVIEW

**Comment by dzen** [comment]
Are you using no_wait with aioamqp ?

**Comment by dzen** [comment]
beware of the typo `on_wait` in your commit message :p

**Comment by foolswood** [comment]
I am not using no_wait, it was just the visual noise in the code (and the open issue) prompting a friday afternoon tidy.

**Comment by dzen** [comment]
Happy to know we have a few tests ;)

**Comment by foolswood** [comment]
I was leaning on them for this, they did their job, let's see if that fix makes things any better.

**Comment by foolswood** [comment]
Seems fine now, fixed the typo whilst at it.

File: aioamqp/channel.py [hunk 1]
```diff
@@ -188,21 +188,32 @@ def server_channel_close(self, frame):
         self.connection_closed(results['reply_code'], results['reply_text'])
 
     @asyncio.coroutine
+    def _write_frame_awaiting_response(
+            self, waiter_id, frame, request, no_wait, **kwargs):
```
Comment (dzen) [inline aioamqp/channel.py]: This indentation seems really odd

**Comment by foolswood** [comment]
I believe this is ready now.

**Comment by dzen** [comment]
Yes, squash both commits and I'll merge it

**Comment by foolswood** [comment]
They're squashed.

**Comment by dzen** [comment]
It's merged since yesterday.

**Pull request closed without merging**